src/javascript/JContent/ContentRoute/ContentLayout/Upload/UploadItem/UploadItem.jsx

```diff
@@ -179,49 +177,20 @@ export class UploadItem extends React.Component {
     handleUpload(type) {
         const {file, path, client} = this.props;
         if (type === 'import') {
-            return client.mutate({
-                mutation: importContent,
-                variables: {
-                    path: `${path}`,
-                    fileHandle: file
-                }
-            });
+            return registry.get('fileUpload', 'import').handleUpload({path, file, client});
         }
 
         if (type === 'replace') {
-            return client.mutate({
-                mutation: updateFileContent,
-                variables: {
-                    path: `${path}/${this.getFileName()}`,
-                    mimeType: file.type,
-                    fileHandle: file,
-                    isImage: isImageFile(file.name)
-                }
-            });
+            let newPath = `${path}/${this.getFileName()}`;
+            return registry.get('fileUpload', 'replace').handleUpload({path: newPath, file, client});
         }
 
         if (type === 'replaceWith') {
-            return client.mutate({
-                mutation: updateFileContent,
-                variables: {
-                    path: `${path}`,
-                    mimeType: file.type,
-                    fileHandle: file,
-                    isImage: isImageFile(file.name)
-                }
-            });
+            return registry.get('fileUpload', 'replace').handleUpload({path, file, client});
         }
 
-        return client.mutate({
-            mutation: uploadFile,
-            variables: {
-                fileHandle: file,
-                nameInJCR: this.getFileName(),
-                path: path,
-                mimeType: file.type,
-                isImage: isImageFile(file.name)
-            }
-        });
```

Comment:
As proposed earlier use something more generic, and add the `canHandle` into registry entries :  
```suggestion
        return registry.find({type: 'fileUpload'}).find(upload => upload.canHandle(type, this.props)).handleUpload(type, this.props);
```

Reply:
you can also make it even more simple by using : 
```return registry.get('fileUpload', type).handleUpload(this.props);``` , where the name of your registry entry would match the type

Reply:
I actually checked this portion to see if I can re-factor this into the one you are currently specifying.
There are two limiting factors:
1. For regular upload the **type** value is actually "undefined" (I don't believe it is a good idea to register a type of "undefined"
2. Existing code has this functionality to get specific filename on "replace" and "undefined" based on state variables in `this.getFileName()`. Props doesn't have this value so need some special handling.

Reply:
I think that's ok for now, we may want to improve that later. Quite a shame we have this `getFilename()` stuff here.